Formula/e/emqx.rb

```diff
@@ -38,20 +37,37 @@ class Emqx < Formula
 
   def install
     ENV["PKG_VSN"] = version.to_s
+    ENV["BUILD_WITHOUT_QUIC"] = "1"
     touch(".prepare")
-    system "make", "emqx"
+    system "make", "emqx-rel"
     prefix.install Dir["_build/emqx/rel/emqx/*"]
     %w[emqx.cmd emqx_ctl.cmd no_dot_erlang.boot].each do |f|
       rm bin/f
     end
     chmod "+x", prefix/"releases/#{version}/no_dot_erlang.boot"
     bin.install_symlink prefix/"releases/#{version}/no_dot_erlang.boot"
+    if OS.mac?
+      # ensure load path for libcrypto is correct
+      crypto_vsn = `erl -noshell -eval 'io:format("~s", [crypto:version()]), halt().'`.strip
+      libcrypto = Formula["openssl@3"].opt_lib/"libcrypto.3.dylib"
+      new_libcrypto = Pathname.new(libcrypto).realpath.to_s
+      %w[crypto.so otp_test_engine.so].each do |f|
+        dynlib = MachO::MachOFile.new(lib/"crypto-#{crypto_vsn}/priv/lib/#{f}")
+        old_libcrypto = dynlib.linked_dylibs.find { |d| d.include?("libcrypto.3.dylib") }
+        next if old_libcrypto.nil?
+
+        dynlib.change_install_name(old_libcrypto, new_libcrypto)
+        dynlib.write("#{dynlib.filename}.1")
+        mv "#{dynlib.filename}.1", dynlib.filename
+        MachO.codesign!(dynlib.filename) if Hardware::CPU.arm?
+      end
+    end
```

Comment:
Let's make this a bit simpler:
```suggestion
    return unless OS.mac?

    # ensure load path for libcrypto is correct
    crypto_vsn = Utils.safe_popen_read("erl", "-noshell", "-eval", 'io:format("~s", [crypto:version()]), halt().').strip
    libcrypto = Formula["openssl@3"].opt_lib/shared_library("libcrypto", "3")
    %w[crypto.so otp_test_engine.so].each do |f|
      dynlib = lib/"crypto-#{crypto_vsn}/priv/lib"/f
      old_libcrypto = dynlib.dynamically_linked_libraries
                            .find { |d| d.end_with?(libcrypto.basename) }
      next if old_libcrypto.nil?

      dynlib.change_install_name(old_libcrypto, libcrypto.to_s)
      MachO.codesign!(dynlib) if Hardware::CPU.arm?
    end
```
You may have to split some lines for being too long.

Reply:
@carlocab thanks, applied the suggestion. However, the build fails now.

Locally I see this error:
```
Error: An exception occurred within a child process:
  Errno::EACCES: Permission denied @ rb_sysopen - /opt/homebrew/Cellar/emqx/5.1.4/lib/crypto-5.1.4/priv/lib/crypto.so
```
It happens on the `change_install_name` step - that's why I added write to another file and `mv` in my version.